Production_scripts/Process_extracts/Produce_extract_tests/extract_chi_ltc_tests.R

```diff
@@ -0,0 +1,93 @@
+duplicates <- new_data %>%
+  dtplyr::lazy_dt() %>%
+  dplyr::group_by(chi) %>%
+  dplyr::mutate(duplicate_chi = dplyr::n() > 1) %>%
+  dplyr::ungroup() %>%
+  dplyr::group_by(chi, postcode) %>%
+  dplyr::mutate(duplicate_chi_postcode = dplyr::n() > 1) %>%
+  dplyr::ungroup() %>%
+  dplyr::summarise(dplyr::across(c("duplicate_chi", "duplicate_chi_postcode"), sum)) %>%
+  dplyr::ungroup() %>%
+  tibble::as_tibble() %>%
+  tidyr::pivot_longer(
+    cols = tidyselect::everything(),
+    names_to = "measure",
+    values_to = "value"
+  )
+
+# Flag when a person has an LTC but no diagnosis date
+valid_dates <- new_data %>%
+  dplyr::mutate(
+    arth_valid = (arth > 0 & !is.na(arth_date)),
+    asthma_valid = (asthma > 0 & !is.na(asthma_date)),
+    atrialfib_valid = (atrialfib > 0 & !is.na(atrialfib_date)),
+    cancer_valid = (cancer > 0 & !is.na(cancer_date)),
+    cvd_valid = (cvd > 0 & !is.na(cvd_date)),
+    liver_valid = (liver > 0 & !is.na(liver_date)),
+    copd_valid = (copd > 0 & !is.na(copd_date)),
+    dementia_valid = (dementia > 0 & !is.na(dementia_date)),
+    diabetes_valid = (diabetes > 0 & !is.na(diabetes_date)),
+    epilepsy_valid = (epilepsy > 0 & !is.na(epilepsy_date)),
+    chd_valid = (chd > 0 & !is.na(chd_date)),
+    hefailure_valid = (hefailure > 0 & !is.na(hefailure_date)),
+    ms_valid = (ms > 0 & !is.na(ms_date)),
+    parkinsons_valid = (parkinsons > 0 & !is.na(parkinsons_date)),
+    refailure_valid = (refailure > 0 & !is.na(refailure_date)),
+    congen_valid = (congen > 0 & !is.na(congen_date)),
+    bloodbfo_valid = (bloodbfo > 0 & !is.na(bloodbfo_date)),
+    endomet_valid = (endomet > 0 & !is.na(endomet_date)),
+    digestive_valid = (digestive > 0 & !is.na(digestive_date))
+  ) %>%
+  dplyr::summarise(
+    dplyr::across(dplyr::contains("valid"), ~ sum(.x, na.rm = TRUE)),
+    dplyr::across(c(arth:digestive), sum, na.rm = TRUE)
+  ) %>%
+  dplyr::mutate(
+    arth_invalid = arth - arth_valid,
+    asthma_invalid = asthma - asthma_valid,
+    atrialfib_invalid = atrialfib - atrialfib_valid,
+    cancer_invalid = cancer - cancer_valid,
+    cvd_invalid = cvd - cvd_valid,
+    liver_invalid = liver - liver_valid,
+    copd_invalid = copd - copd_valid,
+    dementia_invalid = dementia - dementia_valid,
+    diabetes_invalid = diabetes - diabetes_valid,
+    epilepsy_invalid = epilepsy - epilepsy_valid,
+    chd_invalid = chd - chd_valid,
+    hefailure_invalid = hefailure - hefailure_valid,
+    ms_invalid = ms - ms_valid,
+    parkinsons_invalid = parkinsons - parkinsons_valid,
+    refailure_invalid = refailure - refailure_valid,
+    congen_invalid = congen - congen_valid,
+    bloodbfo_invalid = bloodbfo - bloodbfo_valid,
+    endomet_invalid = endomet - endomet_valid,
+    digestive_invalid = digestive - digestive_valid
+  ) %>%
+  tidyr::pivot_longer(
+    cols = tidyselect::everything(),
+    names_to = "measure",
+    values_to = "value"
+  )
```

Comment:
While what you have here is good/working as we would expect to return a meaningful output i'm not sure we will need to include this as a test. Perhaps just the invalid dates would be sufficient as a check for our code. From looking at the processing for the LTC script, we actually derive the LTC flags from the dates and they are provided by IT so in theory we should never have a situation where a person has an LTC flag but does not have a date associated with it.

Reply:
Removed this section entirely